### PR TITLE
Add support for DFRobot ESP32 C3

### PR DESCRIPTION
**Pull request by doskoi**
*By completing this PR sufficiently, you help us to review this Pull Request quicker and also help improve the quality of Release Notes*

### Checklist
1. [x] Please provide specific title of the PR describing the change, including the component name (*eg. „Update of Documentation link on Readme.md“*)
2. [x] Please provide related links (*eg. Issue which will be closed by this Pull Request*)
3. [ ] Please **update relevant Documentation** if applicable
4. [x] Please check [Contributing guide](https://docs.espressif.com/projects/arduino-esp32/en/latest/contributing.html)

*This entire section above can be deleted if all items are checked.*

-----------
## Description of Change
Add support for DFRobot ESP32 C3 pinouts

## Tests scenarios
Please describe on what Hardware and Software combinations you have tested this Pull Request and how.

(*eg. I have tested my Pull Request on Arduino-esp32 core v2.0.2 with ESP32 and ESP32-S2 Board with this scenario*)

## Related links
https://wiki.dfrobot.com/SKU_DFR0868_Beetle_ESP32_C3

(*eg. Closes #number of issue*)

### CODE REVIEW

**Comment by me-no-dev** [comment]
This pull request is incomplete. It is missing the board definition in `boards.txt`

File: boards.txt [hunk 1]
```diff
@@ -14751,3 +14751,227 @@ aw2eth.menu.DebugLevel.verbose=Verbose
...
+beetle_esp32_c3.upload.wait_for_upload_port=false
+
+beetle_esp32_c3.serial.disableDTR=false
+beetle_esp32_c3.serial.disableRTS=false
+
+beetle_esp32_c3.build.tarch=riscv32
+beetle_esp32_c3.build.target=esp
+beetle_esp32_c3.build.mcu=esp32c3
+beetle_esp32_c3.build.core=esp32
+beetle_esp32_c3.build.variant=beetle_esp32_c3
```
Comment (atanisoft) [inline boards.txt:14892]: I believe this value should match the directory for the pins_arduino.h file.

**Comment by doskoi** [comment]
Updated the commits with rebase to make it able to merge

File: boards.txt (hunk 1 above)
Comment (me-no-dev) [reply boards.txt:14780]: yes. it should be `beetle32c3`

**Comment by me-no-dev** [comment]
Could you please sync with this PR as well: https://github.com/espressif/arduino-esp32/pull/6869

Which one should be merged? It seems they cover the same board.

File: boards.txt (hunk 1 above)
Comment (doskoi) [reply boards.txt]: thanks to point it, fixed

**Comment by Arya11111** [comment]
> > Could you please sync with this PR as well: #6869
> > Which one should be merged? It seems they cover the same board.
> 
> I've rebase and cherry-pick the commits again. the Files changes looks good now

Hello, I am the developer of DFRobot, we have added the Beetle C3 board to the arduino-esp32 SDK, you can use the one we added, can you close this PR?

**Comment by Arya11111** [comment]
> > Could you please sync with this PR as well: #6869
> > Which one should be merged? It seems they cover the same board.
> 
> I've rebase and cherry-pick the commits again. the Files changes looks good now

Our DFRobot official has submitted a PR of beetle esp32c3 to arduino-esp32, you can check our frok library: https://github.com/DFRobot/arduino-esp32, we hope you can close the PR you submitted, thank you.

**Comment by Arya11111** [comment]
> > > Could you please sync with this PR as well: #6869
> > > Which one should be merged? It seems they cover the same board.
> > 
> > 
> > I've rebase and cherry-pick the commits again. the Files changes looks good now
> 
> Our DFRobot official has submitted a PR of beetle esp32c3 to arduino-esp32, you can check our frok library: https://github.com/DFRobot/arduino-esp32, we hope you can close the PR you submitted, thank you.

https://github.com/DFRobot/arduino-esp32/tree/master/variants/dfrobot_beetle_esp32c3

**Comment by Arya11111** [comment]
> > Our DFRobot official has submitted a PR of beetle esp32c3 to arduino-esp32, you can check our frok library: https://github.com/DFRobot/arduino-esp32, we hope you can close the PR you submitted, thank you.
> https://github.com/DFRobot/arduino-esp32/tree/master/variants/dfrobot_beetle_esp32c3

**Comment by Arya11111** [comment]
> Our DFRobot official has submitted a PR of beetle esp32c3 to arduino-esp32, you can check our frok library: https://github.com/DFRobot/arduino-esp32, we hope you can close the PR you submitted, thank you.
> https://github.com/DFRobot/arduino-esp32/tree/master/variants/dfrobot_beetle_esp32c3

Our DFRobot official has submitted a PR of beetle esp32c3 to arduino-esp32, you can check our frok library: https://github.com/DFRobot/arduino-esp32, we hope you can close the PR you submitted, thank you.
https://github.com/DFRobot/arduino-esp32/tree/master/variants/dfrobot_beetle_esp32c3

**Comment by Arya11111** [comment]
> > Could you please sync with this PR as well: #6869
> > Which one should be merged? It seems they cover the same board.
> 
> I've rebase and cherry-pick the commits again. the Files changes looks good now

Our DFRobot official has submitted a PR of beetle esp32c3 to arduino-esp32, you can check our frok library: https://github.com/DFRobot/arduino-esp32, we hope you can close the PR you submitted, thank you.
https://github.com/DFRobot/arduino-esp32/tree/master/variants/dfrobot_beetle_esp32c3

**Comment by doskoi** [comment]
@Arya11111  No problem.

**Pull request closed without merging**